test/blackbox-tests/test-cases/config/config-version.t

```diff
@@ -0,0 +1,56 @@
+Tests verifying that the config file presents accurate error or warning
+messages when a stanza is used with a version of dune which does not support
+such a stanza.
+
+Create a config file to use in the following tests. We will initialize the 
+config with a version and a stanza incompatable with that version.
+
+  $ touch dune-config
+  $ cat >dune-config <<EOF
+  > (lang dune 1.0)
+  > (cache enabled)
+  > EOF
+
+Attempt to initialize a new project while an invaild stanza due to versioning
+exists in the config.
+
+  $ dune init proj test --config-file=dune-config
+  File "$TESTCASE_ROOT/dune-config", line 2, characters 0-15:
+  2 | (cache enabled)
+      ^^^^^^^^^^^^^^^
+  Error: 'cache' is only available since version 2.0 of the dune language.
+  Please update your dune config file to have (lang dune 2.0).
+  [1]
```

Comment:
Can you add a test that adds `(lang dune 2.0)` and shows this works? There is probably a test that makes sure of it, but it is also useful to show that it works.